File: tests/benchmarks/primes/primes.js
function isPrime(n) {
	if (n < 2) return false;
	for (var i=2; i < n; i++) {
		if (n % i === 0) return false;
	}
	return true;
}

var start = process.hrtime();

var i=0, N=30000, primes=[];
for (; i < N; i++) {
	if (isPrime(i)) {
		primes.push(i);
	}
}

var end = process.hrtime(start);
var secs = (end[0] + end[1] / 1e9).toFixed(6) + 's';
console.log('elapsed:', secs);

File: tests/benchmarks/primes/primes.lua
local function isPrime(n)
	if n < 2 then
		return false
	end
  for i=2, n-1 do
    if n % i == 0 then
			return false
		end
  end
  return true
end

local start = os.clock()

local N = 30000
local primes = {}

for i = 0, N - 1 do
	if isPrime(i) then
		table.insert(primes, i)
	end
end

local seconds = os.clock() - start
print('elapsed: ' .. seconds .. 's')
